plugins/hypervisors/kvm/src/com/cloud/ha/KVMInvestigator.java

```diff
@@ -40,17 +40,23 @@
 public class KVMInvestigator extends AdapterBase implements Investigator {
     private final static Logger s_logger = Logger.getLogger(KVMInvestigator.class);
     @Inject
-    HostDao _hostDao;
+    private HostDao _hostDao;
+    @Inject
+    private AgentManager _agentMgr;
     @Inject
-    AgentManager _agentMgr;
+    private ResourceManager _resourceMgr;
     @Inject
-    ResourceManager _resourceMgr;
+    private PrimaryDataStoreDao _storagePoolDao;
     @Inject
-    PrimaryDataStoreDao _storagePoolDao;
+    private HAManager haManager;
 
     @Override
     public boolean isVmAlive(com.cloud.vm.VirtualMachine vm, Host host) throws UnknownVM {
+        if (haManager.isHAEligible(host)) {
```

Comment:
@koushik-das FYI - injections here for VM-HA Host-HA coordination, the eligibility checks several things -- including that the feature is explicitly enabled for a host.